File: 3890.IntegersWithMultipleSumofTwoCubes.cpp
//first submission

class Solution {
public:
    vector<int> findGoodIntegers(int n) {
        unordered_map<int,int> cubes;
        for(int i = 1; i*i*i<=n; i++){
            for(int j = i; j*j*j + i*i*i <=n; j++){
                int x = j*j*j + i*i*i;
                cubes[x]++;
            }
        }
        vector<int>ans;
        for(auto &it : cubes){
            if(it.second>=2){
                ans.push_back(it.first);
            }
        }
        sort(ans.begin(), ans.end());
        return ans;
    }
};

File: 3890.IntegersWithMultipleSumofTwoCubes.py
//first submission
class Solution:
    def findGoodIntegers(self, n: int) -> list[int]:
        cubes = {}

        i = 1

        while i**3<=n:
            j = i
            while i**3 + j**3 <= n:
                x = j**3 + i**3
                cubes[x] = cubes.get(x, 0) + 1
                j += 1

            i += 1
        
        ans = []

        for key, value in cubes.items():
            if value>=2:
                ans.append(key)
        
        ans.sort()
        return ans